From 316ccb1733a6da726c0e7f0748e3e88ec459ca54 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?An=C3=ADbal=20Lim=C3=B3n?= <anibal.limon@linux.intel.com>
Date: Fri, 13 May 2016 14:36:51 -0500
Subject: [PATCH] src/{dialog,gtkprint}.c: Fix security formatting issues
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

../../../../../../../workspace/sources/leafpad/src/dialog.c: In
unction
'run_dialog_message':
../../../../../../../workspace/sources/leafpad/src/dialog.c:39:3:
rror:
format not a string literal and no format arguments
[-Werror=format-security]
   str);
   ^
../../../../../../../workspace/sources/leafpad/src/dialog.c: In
unction
'create_dialog_message_question':
../../../../../../../workspace/sources/leafpad/src/dialog.c:64:3:
rror:
format not a string literal and no format arguments
[-Werror=format-security]
   str);

../../../../../../../workspace/sources/leafpad/src/gtkprint.c: In
function 'create_error_dialog':
../../../../../../../workspace/sources/leafpad/src/gtkprint.c:168:3:
error: format not a string literal and no format arguments
[-Werror=format-security]
   message);

[YOCTO #9546]

Signed-off-by: Aníbal Limón <anibal.limon@linux.intel.com>

Upstream-status: Pending
---
 src/dialog.c   | 4 ++--
 src/gtkprint.c | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/dialog.c b/src/dialog.c
index 14b69d7..8c8f2da 100644
--- a/src/dialog.c
+++ b/src/dialog.c
@@ -36,7 +36,7 @@ void run_dialog_message(GtkWidget *window,
 		GTK_DIALOG_DESTROY_WITH_PARENT,
 		type,
 		GTK_BUTTONS_NONE,
-		str);
+		str, NULL);
 	gtk_window_set_resizable(GTK_WINDOW(dialog), FALSE);
 	gtk_dialog_add_buttons(GTK_DIALOG(dialog),
 		GTK_STOCK_OK, GTK_RESPONSE_CANCEL, NULL);
@@ -61,7 +61,7 @@ GtkWidget *create_dialog_message_question(GtkWidget *window, gchar *message, ...
 		GTK_DIALOG_DESTROY_WITH_PARENT,
 		GTK_MESSAGE_QUESTION,
 		GTK_BUTTONS_NONE,
-		str);
+		str, NULL);
 	gtk_window_set_resizable(GTK_WINDOW(dialog), FALSE);
 	gtk_dialog_add_buttons(GTK_DIALOG(dialog),
 		GTK_STOCK_NO, GTK_RESPONSE_NO,
diff --git a/src/gtkprint.c b/src/gtkprint.c
index 3f39384..e2bb83a 100644
--- a/src/gtkprint.c
+++ b/src/gtkprint.c
@@ -165,7 +165,7 @@ static void create_error_dialog(GtkTextView *text_view, gchar *message)
 		GTK_DIALOG_DESTROY_WITH_PARENT,
 		GTK_MESSAGE_ERROR,
 		GTK_BUTTONS_NONE,
-		message);
+		message, NULL);
 	gtk_window_set_resizable(GTK_WINDOW(dialog), FALSE);
 	gtk_dialog_add_buttons(GTK_DIALOG(dialog),
 		GTK_STOCK_OK, GTK_RESPONSE_CANCEL, NULL);
-- 
2.1.4

